### PR TITLE
Move //third_party/glfw to //flutter/third_party/glfw

### PR DESCRIPTION
**Pull request by cbracken**
As part of eliminating the Flutter buildroot (#67373), we are moving all third-party dependencies from //third_party to //flutter/third_party.

This is the engine-side follow-up to flutter/buildroot#777.

Once all third-party dependencies have been migrated, tooling and config will be moved and the buildroot will be eliminated altogether.

Issue: https://github.com/flutter/flutter/issues/136284

No tests changed because there is no semantic change to this PR. This is simply relocating a dependency.

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
The initial commit will fail the licence script; once we get the failures from CI, I'll amend the patch.

**Review by matanlurey — APPROVED** [review]
FREAKIN LOVE IT

File: .gitignore [hunk 1]
```diff
@@ -28,6 +28,8 @@ pubspec.lock
 docs/doxygen/
 xcuserdata
 
+# directories pulled in via deps or hooks
+third_party/glfw/
```
Comment (zanderso) [inline .gitignore:32]: Is this needed after https://github.com/flutter/engine/pull/46729 ?

**Review by zanderso — APPROVED** [review]
lgtm w/ question

File: .gitignore (hunk 1 above)
Comment (cbracken) [reply .gitignore:32]: Nope -- good catch!

Comment (matanlurey) [reply .gitignore]: Nope, we fixed it!

**Comment by matanlurey** [comment]
Latest build error:

> error searching for copyright in: ../../../flutter/third_party/glfw/deps/getopt.h
> failed to find any license but saw unmatched potential copyright and license statements; first twenty lines

@cbracken see https://raw.githubusercontent.com/flutter/engine/main/ci/licenses_golden/licenses_third_party, but I think we'll need to move these licenses to https://github.com/flutter/engine/blob/main/ci/licenses_golden/licenses_flutter (the names are a bit of a misnomer).

**Comment by zanderso** [comment]
It might not fix everything the license script is sad about, but I would suggest starting out by updating the paths here:

https://github.com/flutter/engine/blob/main/tools/licenses/lib/paths.dart#L101

and seeing what the remaining issues are.

**Pull request merged**